src/calculation/CompositeReportBuilder.ts

```diff
@@ -103,29 +118,91 @@ export class CompositeReportBuilder<T extends PopulationGroupResult> extends Abs
 
   private addWeightedResults(results: ExecutionResult<T>[]) {
     // https://build.fhir.org/ig/HL7/cqf-measures/composite-measures.html#weighted-scoring
-    const componentPopulationResults: Record<string, { numerator: number; denominator: number; weight: number }> = {};
+    // map component canonical to its population results (if only one group exists), or map to a Record that
+    // maps each component group to its corresponding weight
+    const componentPopulationResults: Record<
+      string,
+      ComponentPopulationResults | Record<string, ComponentPopulationResults>
+    > = {};
 
     results.forEach(result => {
-      result.componentResults?.forEach(componentResult => {
-        if (componentResult.componentCanonical && !componentPopulationResults[componentResult.componentCanonical]) {
-          componentPopulationResults[componentResult.componentCanonical] = {
-            numerator: 0,
-            denominator: 0,
-            weight: this.weightedComponents[componentResult.componentCanonical]
-          };
-        }
-
-        if (componentResult.componentCanonical && componentPopulationResults[componentResult.componentCanonical]) {
-          if (
-            componentResult.populationResults?.find(pr => pr.populationType === PopulationType.NUMER)?.result === true
-          ) {
-            componentPopulationResults[componentResult.componentCanonical].numerator++;
+      const componentResults = filterComponentResults(this.components, result.componentResults);
+      componentResults.forEach(componentResult => {
+        if (componentResult.componentCanonical) {
+          // component info will either consist of mapping of groups to weights, or a single weight
+          const componentInfo = this.components[componentResult.componentCanonical];
+          // if only one weight is specified, the component corresponds to a single group
+          const hasSingleGroup = typeof componentInfo === 'number';
+          if (hasSingleGroup) {
+            if (!(componentResult.componentCanonical in componentPopulationResults)) {
```

Comment:
hmm what would the error be that we would throw in this case?
Here, if the group id extension is not defined, then we just want to check if the component canonical's weight has already been captured in the `componentPopulationResults` record or not. If not, we add the canonical as a key and map it to the weight. Otherwise, we do not do anything since we already have the weight info for the component, and we can continue with incrementing the numer and denom. 

The reason why we could come across the canonical more than once is because we are looping over the Execution results, meaning we have results for this canonical for each patient. In the screenshot below, you can see that when looping over each patient, we run across the `measure-ComponentTwo` canonical. We only add it as key to the record the first time that we encounter it.
<img width="437" alt="Screenshot 2023-06-05 at 8 48 45 AM" src="https://github.com/projecttacoma/fqm-execution/assets/87094479/75721506-8228-4ef2-a7af-dabdc0edd7d3">

Theoretically, the way that calculation is set up, I would assume that each patient has component results for all the components (unless they are filtered out beforehand based on the groupid ext, which in this case is not present). I guess, rather than throwing an error, we could have a separate loop that initializes the `componentPopulationResults` record so that we do not need to loop through all the records unnecessarily? I kept the initial structure that was used for the weighted scoring and tailored it to the new groupid functionality, which may no longer be the best approach

Reply:
In the above screenshot, I added `console.log`s to lines 129, 138, and 139 in `CompositeReportBuilder.ts` if you want to replicate